main_test.go

```diff
@@ -52,3 +55,112 @@ func TestGetRetryInfo(t *testing.T) {
 		}
 	}
 }
+
+func TestTransformExportJSONRecord(t *testing.T) {
+	testCases := []struct {
+		tableColumns []string
+		rec          map[string]interface{}
+		expResult    []string
+	}{
+		// no custom vars
+		{
+			tableColumns: []string{"eventtargettext", "pageduration", "customvars"},
+			rec: map[string]interface{}{
+				"EventTargetText": "Heyo!",
+				"PageDuration":    42,
+			},
+			expResult: []string{"Heyo!", "42", `{}`},
+		},
+		// two custom vars
+		{
+			tableColumns: []string{"eventtargettext", "pageduration", "customvars"},
+			rec: map[string]interface{}{
+				"EventTargetText": "Heyo!",
+				"PageDuration":    42,
+				"custom_str":      "Heyo again!",
+				"custom_num":      42,
+			},
+			expResult: []string{"Heyo!", "42", `{"custom_str":"Heyo again!","custom_num":42}`},
+		},
+		// missing column value for pageduration
+		{
+			tableColumns: []string{"eventtargettext", "pageduration", "customvars"},
+			rec: map[string]interface{}{
+				"EventTargetText": "Heyo!",
+			},
+			expResult: []string{"Heyo!", "", `{}`},
+		},
+		// additional columns in target table that are not in the export
+		{
+			tableColumns: []string{"eventtargettext", "pageduration", "customvars", "randomcolumnnotinexport"},
+			rec: map[string]interface{}{
+				"EventTargetText": "Heyo!",
+				"PageDuration":    42,
+			},
+			expResult: []string{"Heyo!", "42", `{}`, ""},
+		},
+	}
+
+	for i, tc := range testCases {
+		wh := StubWarehouse{}
+		result, err := TransformExportJSONRecord(&wh, tc.tableColumns, tc.rec)
+		if err != nil {
+			t.Errorf("Unexpected err %s on test case %d", err, i)
+			continue
+		}
+		if len(result) != len(tc.expResult) {
+			t.Errorf("Incorrect length of result; expected %d, got %d on test case %d", len(result), len(tc.expResult), i)
+			continue
+		}
+		for j := range tc.expResult {
+			if !compareTransformedStrings(t, tc.expResult[j], result[j]) {
+				t.Errorf("Result mismatch; expected %s, got %s on test case %d, item %d", tc.expResult[j], result[j], i, j)
+			}
+		}
+	}
+}
+
+func compareTransformedStrings(t *testing.T, str1, str2 string) bool {
+	if str1 == str2 {
+		return true
+	}
+	if len(str1) > 0 && str1[0] == '{' {
+		return compareJSONStrings(t, str2, str2)
+	}
+	return false
+}
+
+func compareJSONStrings(t *testing.T, str1, str2 string) bool {
```

Comment:
I think this is fine for now, but just FYI, we have a `fs/fsjson` package with `Equals()` function.

Reply:
Is that a package there we're keeping private to the primary repo? Remember this is the public hauser repo.